cloudformation/external-slurmdbd/resources/user_data.sh

```diff
@@ -7,13 +7,22 @@ function vendor_cookbook
   tar -xzf /etc/chef/aws-parallelcluster-cookbook.tgz
   HOME_BAK="${!HOME}"
   export HOME="/tmp"
-  for d in `ls /tmp/cookbooks`; do
-    cd /tmp/cookbooks/$d
+  for d in /tmp/cookbooks/*; do
+    cd "$d" || continue
     LANG=en_US.UTF-8 /opt/cinc/embedded/bin/berks vendor /etc/chef/cookbooks --delete
   done;
   export HOME="${!HOME_BAK}"
 }
 
+TOKEN=`curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600"`
+INSTANCE_ID=$(curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/instance-id)
+AVAIL_ZONE=$(curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/placement/availability-zone)
+ENI_ID=$(aws ec2 describe-instances --instance-ids ${!INSTANCE_ID} --region ${Region} | jq .Reservations[0].Instances[0].NetworkInterfaces[0].NetworkInterfaceId | tr -d '"')
+
+aws ec2 assign-private-ip-addresses --region ${Region} --network-interface-id ${!ENI_ID} --private-ip-addresses ${PrivateIp} --allow-reassignment
+
```

Comment:
Will failures be logged or do we need some extra check to detect failures and log them ?

Reply:
At the moment they would just be logged in the cloud-init output. I will create an internal task to make sure we address this.